app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -0,0 +1,85 @@
+package ar.com.wolox.android.example.ui.home.news
+
+import android.support.v7.widget.LinearLayoutManager
+import android.view.View
+import ar.com.wolox.android.R
+import ar.com.wolox.android.example.model.New
+import ar.com.wolox.android.example.utils.InfiniteScroll
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import ar.com.wolox.wolmo.core.util.ToastFactory
+import kotlinx.android.synthetic.main.fragment_news.*
+import javax.inject.Inject
+
+/**
+ * Fragment for news feature;
+ */
+class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INewsView {
+
+    @Inject lateinit var mNewsAdapter: NewsAdapter
+    @Inject lateinit var mToastFactory: ToastFactory
+
+    val TAG = NewsFragment::class.java.simpleName
+    private val VIEW_THRESHOLD = 10
+
+    override fun init() {
+        val layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
+        mNewsList.layoutManager = layoutManager
+        mNewsList.adapter = mNewsAdapter
+    }
+
+    override fun setUi(v: View?) {
+        super.setUi(v)
+        mRefreshNewsSwipe.setColorSchemeResources(R.color.colorPrimary, R.color.colorPrimaryDark, R.color.colorAccent)
+
+        mCreateNewButton.attachToRecyclerView(mNewsList)
+    }
+
+    override fun layout() = R.layout.fragment_news
+
+    override fun setListeners() {
+        super.setListeners()
+
+        mRefreshNewsSwipe.setOnRefreshListener {
+            showRefreshing()
+            presenter.requestNews(paginated = false)
+        }
+
+        mNewsList.addOnScrollListener(object : InfiniteScroll({
+            showRefreshing()
+            presenter.requestNews(paginated = true)
+        }) {
+            override fun hasMore(): Boolean {
+                val layoutManager = mNewsList.layoutManager as LinearLayoutManager
+                val lastVisibleItem = layoutManager.findLastVisibleItemPosition()
+                return (presenter.hasMore()) && (lastVisibleItem >= (layoutManager.itemCount - VIEW_THRESHOLD))
+            }
+
+            override fun isLoading(): Boolean = mRefreshNewsSwipe.isRefreshing
+        })
+    }
+
+    override fun populate() {
+        super.populate()
+        presenter.requestNews(paginated = false)
+    }
+
+    override fun showRefreshing() {
+        mRefreshNewsSwipe?.isRefreshing = true
```

Comment:
There's a reason to say the `mRefreshNewsSwipe` could be null? Why you put the `?`?

Reply:
I thought maybe an update could happen when the view is in a detaching state